### PR TITLE
Das 1867

### PR DESCRIPTION
**Pull request by lyonthefrog**
## Description

This PR encompasses an update to the `/tests/trajectory-subsetter/utilities.py` function, `compare_results_to_reference_file()`. We made this update to make these tests compatible with more Data Access Services cloud collections, as we discovered collections that have a hierarchical data structure. These collections are compatible with [xarray Datetree](https://xarray-datatree.readthedocs.io/en/latest/) but not with xarray Dataset. 

The comparison function was likewise updated to [DataTree.identical()](https://xarray-datatree.readthedocs.io/en/latest/generated/datatree.DataTree.identical.html) which increases the scope of the comparison to check dataset, variable, and coordinate attributes.  

The reference files for the Trajectory Subsetter tests were updated due to typo within the metadata attributes of the GEDI L4A input file (caught by the updated comparison function, DataTree.identical()). This typo is present in the older file, but not present in the new file.

Additionally, the updates to the non-trajectory-subsetter notebooks were just to make sure each service notebook references an environment with the new`<service-image-name>-tests` naming convention. 

## Jira Issue ID
[DAS-1867](https://bugs.earthdata.nasa.gov/browse/DAS-1867)

## Local Test Steps

**1. Run success cases.**

Pull this branch and execute the `TrajectorySubsetter_Regression.ipynb` notebook. Each test should pass.

**2. Run failure cases.**

Edit the arguments for `compare_results_to_reference_file()` in each test to reference the wrong file. Each of these tests should fail.

## PR Acceptance Checklist
* [X] Acceptance criteria met
* [X] Tests added/updated (if needed) and passing
* [X] Documentation updated (if needed)

### CODE REVIEW

File: test/trajectory-subsetter/utilities.py [hunk 1]
```diff
@@ -7,28 +7,26 @@
...
+                                                    'do not match.')
 
-    reference_data.close()
-    results_data.close()
     reference_data = None
     results_data = None
-
-
+    
+
```
Comment (flamingbear) [inline test/trajectory-subsetter/utilities.py:29]: Looks like some wild/unintended whitespace here also end of line 15, and maybe on L24?

**Review by flamingbear — APPROVED** [review]
This looks good to me.  Thanks for updating all of the extra files.  There is just a nit with some whitespace on the utilities file.  But I was able to run the tests and see them pass and fail.  I like the testing entire file instead of just the band.  Nice!

File: test/trajectory-subsetter/utilities.py [hunk 2]
```diff
@@ -7,28 +7,26 @@
...
+                                                    'do not match.')
 
-    reference_data.close()
-    results_data.close()
     reference_data = None
     results_data = None
-
-
+ 
+
```
Comment (owenlittlejohns) [inline test/trajectory-subsetter/utilities.py:29]: (Previous comment seems to have disappeared somewhere...)

Nit: Whitespace in blank line. Running something like `pylint` or `pycodestyle` over this file would catch stuff like this.

**Review by owenlittlejohns — COMMENTED** [review]
Sorry to not finish this review yesterday evening. Overall this looks great and makes sense. I only had super minor nits (more specific imports and some whitespace).

Generally, even though you are updating the prose in a bunch of notebooks, I think you've done the right thing to not iterate their versions, as they haven't materially changed in a way that impacts the tests.

File: test/trajectory-subsetter/utilities.py (hunk 2 above)
Comment (owenlittlejohns) [reply test/trajectory-subsetter/utilities.py:29]: (And that comment reappeared on publishing the review. Weird)

File: test/trajectory-subsetter/TrajectorySubsetter_Regression.ipynb [hunk 3]
```diff
@@ -247,7 +253,11 @@
    "cell_type": "code",
    "execution_count": null,
    "id": "ffb9742c",
-   "metadata": {},
+   "metadata": {
+    "tags": [
+     "raises-exception"
```
Comment (owenlittlejohns) [inline test/trajectory-subsetter/TrajectorySubsetter_Regression.ipynb:258]: Again jumping in far too late. Discussed a little bit in Slack.

I think this metadata tag is super cool to allow all cells to run (so an early failure will still let you know whether later tests pass or fail, in order to better understand the scope of failures in the full suite). That said, we still need to make the notebook fail in some way at the end if any tests fail, so things like Bamboo know that there was at least one failed test.

**Pull request merged**